File: week01/mk/TwoSum.go
// Runtime : 7ms / Memory : 4.3 MB

package main

import "fmt"

func main() {
	result := twoSum([]int{-3, 0, 3, 6}, 0)
	fmt.Println(result)
}

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)
	for index, value := range nums {
		firstNum := target - value
		firstNumIndex, exist := numMap[firstNum]
		if exist {
			secondNumIndex := index
			return []int{firstNumIndex, secondNumIndex}
		}
		numMap[value] = index
	}

	return []int{}
}

File: S01/week01/z0/TwoSum.kt
class Solution {
  // Runtime : 222ms
  // Memory : 39.21MB
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val result = intArrayOf(-1,-1)
        for(i in 0..nums.lastIndex) {
            val remain = target - nums[i]
            for(j in i+1..nums.lastIndex) {
                if(nums[j] == remain) {
                    result[0] = i
                    result[1] = j
                    break
                }
            }
        }
        return result
    }
}
